serving/docker/partition/sm_neo_neuron_partition.py

```diff
@@ -432,37 +281,49 @@ def copy_input_files_to_output(self):
                 if os.path.abspath(entry.path) != optimized_model_dir:
                     shutil.move(entry.path, optimized_model_dir)
 
+        # use safetensors index to determine which weights to skip copying
+        ignore_fun = None
+        try:
+            safetensors_index_file = os.path.join(
+                self.INPUT_MODEL_DIRECTORY, "model.safetensors.index.json")
+            # build the set of weight files to ignore
+            weight_files = set()
+            with open(safetensors_index_file) as f:
+                index = json.load(f)
+                for k, v in index["weight_map"].items():
+                    weight_files.add(v)
+            # function passed to copytree that ignores all matching weight files in the
+            # root input directory
+            def _ignore_fun(directory, files):
```

Comment:
Consider abbreviating this into a lambda 1-liner.

Reply:
Intended to do so initially. In this case I feel it's more readable as a standard function definition.